python/mxnet/context.py

```diff
@@ -29,8 +29,8 @@ class Context(object):
     """
     # static class variable
     default_ctx = None
-    devtype2str = {1: 'cpu', 2: 'gpu'}
-    devstr2type = {'cpu': 1, 'gpu': 2}
+    devtype2str = {1: 'cpu', 2: 'gpu', 3: 'cpu_pinned'}
+    devstr2type = {'cpu': 1, 'gpu': 2, 'cpu_pinned': 3}
```

Comment:
must be here, create state may create pinned mem array. That bug costs me a
hour to find

On Sun, Oct 11, 2015 at 1:45 AM Tianqi Chen <notifications@github.com>
wrote:

> In python/mxnet/context.py
> <https://github.com/dmlc/mxnet/pull/263#discussion_r41705789>:
>
> > @@ -29,8 +29,8 @@ class Context(object):
> >      """
> >      # static class variable
> >      default_ctx = None
> > -    devtype2str = {1: 'cpu', 2: 'gpu'}
> > -    devstr2type = {'cpu': 1, 'gpu': 2}
> > +    devtype2str = {1: 'cpu', 2: 'gpu', 3: 'cpu_pinned'}
> > +    devstr2type = {'cpu': 1, 'gpu': 2, 'cpu_pinned': 3}
>
> Let us not expose this for now
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dmlc/mxnet/pull/263/files#r41705789>.
>